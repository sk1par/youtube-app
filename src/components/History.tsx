import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import selectedVideo from '../actions/selectedVideoActions';
import IVideos from '../models/videos.interface';
import { selectHistoryVideos } from '../selectors/selectors';

const History = () => {
    const historyVideos = useSelector(selectHistoryVideos);
    const dispatch = useDispatch();

    const selectVideo = (video: IVideos) => {
        dispatch(selectedVideo(video));
    }

    return (
        <>
        <div className='button'>
            <h1>History</h1>
        </div>

        <div>
            {historyVideos?.length > 0 ? historyVideos.map((video: IVideos) => {
               return  <p onClick={() => selectVideo(video)} className='link' key={video.id.videoId}>{video.snippet.title}</p>
            }) : <p className='button'>No recent history</p>}
        </div>
        </>
    )
}

export default History
