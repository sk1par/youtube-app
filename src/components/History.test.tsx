import History from '../components/History';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallowToJson } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn()
}));

let wrapper: any;

beforeEach(() => {
    wrapper = shallow(<History />);
});

it('renders correctly', () => {
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
