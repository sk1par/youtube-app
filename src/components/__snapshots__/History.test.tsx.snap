// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <div
    className="button"
  >
    <h1>
      History
    </h1>
  </div>
  <div>
    <p
      className="button"
    >
      No recent history
    </p>
  </div>
</Fragment>
`;
